### PR TITLE
Add Partial unification for Clearer Documentation

### PR DESCRIPTION
**Pull request by ChristopherDavenport**
People are running into issues as this is not added to their applications. We should be more transparent about the need to add this.

### CODE REVIEW

**Comment by ChristopherDavenport** [comment]
I believe that is in progress currently.

File: docs/src/main/tut/service.md [hunk 1]
```diff
@@ -22,6 +22,8 @@ libraryDependencies ++= Seq(
   "org.http4s" %% "http4s-blaze-server" % http4sVersion,
   "org.http4s" %% "http4s-blaze-client" % http4sVersion
 )
+
+scalacOptions ++= Seq("-Ypartial-unification")
```
Comment (rossabaker) [inline docs/src/main/tut/service.md]: There's a [plugin](https://github.com/fiadliel/sbt-partial-unification) that covers a few more cases.  Namely, it uses the plugin for versions of Scala that predate the flag.  Would we be better using that?

Comment (rossabaker) [reply docs/src/main/tut/service.md]: The plugin also works on 2.10.6, though that only helps a subset of the subset on Scalaz.  Yeah, I'll go along with this.

I might put in a plug for sbt-tpolecat, which does this _and_ more.

Comment (ChristopherDavenport) [reply docs/src/main/tut/service.md]: I recommend another PR. That seems outside of the scope of this PR and could get some debate.

**Pull request merged**